Release notes to sprint 60

nrichers opened the pull request:
## Internal Notes for Reviewers

This PR includes release notes up to Sprint 60: [**LIVE PREVIEW — October 22, 2024**](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6860/release-notes-to-sprint-60/releases/2024-oct-22/release-notes.html)

### Frontend
- https://github.com/validmind/frontend/releases/tag/v1.24.15
- https://github.com/validmind/frontend/releases/tag/v1.25.0
- https://github.com/validmind/frontend/releases/tag/v1.25.7
- https://github.com/validmind/frontend/releases/tag/v1.25.15
- https://github.com/validmind/frontend/releases/tag/v1.25.19

### Developer framework
- https://github.com/validmind/developer-framework/releases/tag/v2.5.15
- https://github.com/validmind/developer-framework/releases/tag/v2.5.18

### Documentation

- https://github.com/validmind/documentation/releases/tag/v2.5.18

## Other improvements

### New button class for Markdown links

I got tired of our hard-coded HTML buttons and added a new class for buttons to simplify the process. You can now add `{.button}` to any Markdown link to turn it into a button:

<img width="190" alt="image" src="https://github.com/user-attachments/assets/ae5b8561-f180-4b36-819b-2adf0617b60c">

| Before | After |
|----|----------|
| ```<form method="get" action="../../guide/monitoring/ongoing-monitoring.html"><button type="submit" style="color: white; background-color: #de257e; border-radius: 8px; border: none; font-size: 16px; padding: 6.25px 12.5px; margin-left: 16px; margin-bottom: 20px;">Learn about monitoring</button></form>``` | `[Learn about monitoring](../../guide/monitoring/ongoing-monitoring.qmd){.button}`  | 

### Rounded edges for screenshots

Quite subtle, but our class="screenshot" was very rectilinear. I added the same rounding for corners as for the button:

| Before | After |
|----------|----------|
| <img width="481" alt="image" src="https://github.com/user-attachments/assets/d2746082-dc44-4053-989b-88f66adbcf3c"> | <img width="484" alt="image" src="https://github.com/user-attachments/assets/de784229-88db-4b33-93cd-51c24eb627c2"> |

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `deprecation`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

nrichers commented on site/releases/2024-oct-22/release-notes.qmd:
```diff
@@ -0,0 +1,513 @@
+[{{< fa brands github >}} Try it on GitHub](https://github.com/validmind/developer-framework){.button}
+:::
+
+::::
```
TO DO: Our instructions say to add `{{< include /releases/_how-to-upgrade.qmd >}}` at the end, but the file seems to be missing from our repo.

nrichers commented on site/releases/2024-oct-22/release-notes.qmd:
```diff
@@ -0,0 +1,513 @@
+
+Monitoring is a critical component of model risk management, as emphasized in regulations such as SR 11-7, SS1/23, and E-24. With this release of ValidMind, we officially support ongoing monitoring. You can enable this feature for both existing and new models.
+
+![Ongoing monitoring of a model](/releases/2024-oct-22/358433496-02a6d5b9-0521-4110-b45c-b91e06798a97.png){fig-alt="An image showing some ongoing monitoring data on the ValidMind platform" class="screenshot"}
```
@nrichers — reminder for yourself to add `fig-alt` text to the images.

nrichers commented on site/releases/2024-oct-22/release-notes.qmd:
```diff
@@ -0,0 +1,513 @@
+
+Watch the demo:
+
+<div style="position: relative; padding-bottom: 62.5%; height: 0;"><iframe src="https://www.loom.com/embed/4d7d14b21d17416caefddf2b46fe05af?sid=70f2acd6-9a29-4856-a6ed-b3a0b28b084b" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>
```
@gtagle you had created this demo video for custom field management that seems helpful — are you OK with us including this in the release notes?

validbeck reviewed (changes requested):
### UPDATES

<img width="613" alt="Screenshot 2024-10-21 at 9 25 58 AM" src="https://github.com/user-attachments/assets/17ec4173-d28d-44ce-94ba-c81aea493962">

- I've updated the CSS for the buttons.
- I've updated the style guide & README to reflect the proper link & button conventions.

### GLOBAL COMMENTS

- **We've being using `target="_blank` for CTA links out of release notes, FYI.** Can you please add these?
- **I don't like the "Try it..." or "Learn about..." especially without the name of the link, and as we indicate in our style guide, nebulous/clumsy links should be avoided.** For example, this is what I ended up doing in the clean-up: 
<img width="888" alt="Screenshot 2024-10-21 at 8 59 19 AM" src="https://github.com/user-attachments/assets/10378a22-013a-48ec-bfe0-8a518341031c"> (**If there is a user guide, just use the name of the guide! No need for a lead-in.**) 
- **I also don't like the FA icons, too busy.** Please remove them.
- **Video thumbnails could be bigger.** You can steal the dimensions from this release: https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6860/release-notes-to-sprint-60/releases/2024-may-22/release-notes.html#new-quickstart-video
- I would break up some of the more chunkier paragraphs into some bullet points. 
- The `_how-to-upgrade.qmd` includes is in `main`, so you might need to pull in to retrieve it so you can append it at the end.

### Section comments

#### Support for ongoing monitoring

- The screenshot here needs to be cropped ;)
- I'd also do a two column for the image with all the text up to "The monitoring template for your model automatically populates with data as your code runs..." and see what that looks like.  

#### Support for bias and fairness tests

We need some links to the metric test reference pages!

#### New Metric Over Time content block

The three screenshots are two cramped. I'd try two rows:

1. Text | Adding a new block & CTA link (button should be centered)
2. Selecting the block | Metric over time graph

#### Filtering columns when running tests

Needs a link to the `DatasetDescription` reference.

#### Enhanced custom field management with Python support

This video needs the styling we apply to all videos.

#### Rich text editor for validation guidelines

The columns here are squished.

#### Updated training content

- Let's use the listing standard here to make the training stand out a bit instead of the random screenshot link:  https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6860/release-notes-to-sprint-60/releases/2024-may-22/release-notes.html#sandbox-getting-started
- The column widths could also be adjusted to be a bit neater.

nrichers commented:
@validbeck, great feedback, thank you. Will address where you haven't already. Some comments:

> We've being using target="_blank for CTA links out of release notes, FYI. Can you please add these?

I don't think this should be true globally. We should open links in new tabs ONLY if the links are external and this is already configured in _quarto.yml via `link-external-newwindow: true`. Links to other pages in the docs should not be spawning tabs for the docs site all over the place, especially given how many links we now have. Less is more. 

(For reference, at my previous gig we had a rule that internal docs links could never spawn another tab, as we have >300k pages and it was just messy otherwise. You can always open links in a new tab if this is your preference.)

> We need some links to the metric test reference pages!
> Needs a link to the DatasetDescription reference.

I'm on the fence about these but with a preference not to add too many links — release notes should tell you _how to get started_ with something after you read them and that should be the notebooks, not some reference page. Again, less is more.

validbeck commented:
Forgot to mention, I like the new rounded corners! 

> I don't think this should be true globally. We should open links in new tabs ONLY if the links are external and this is already configured in _quarto.yml via link-external-newwindow: true. Links to other pages in the docs should not be spawning tabs for the docs site all over the place, especially given how many links we now have. Less is more.

This is how the buttons were when I grabbed them to do the first round of links — I will adjust in my PR now that I know this wasn't intentional.

> I'm on the fence about these but with a preference not to add too many links — release notes should tell you how to get started with something after you read them and that should be the notebooks, not some reference page. Again, less is more.

I understand what you're arguing, but in this case I vote for them. As a developer, I would expect to see the reference for the function or test you're talking about.

nrichers commented:
> This is how the buttons were when I grabbed them to do the first round of links — I will adjust in my PR now that I know this wasn't intentional.

I think that was likely a side effect of which button you grabbed to create others which just goes to show that there is an issue with the old HTML button approach. Thank you for fixing!

> I understand what you're arguing, but in this case I vote for them. As a developer, I would expect to see the reference for the function or test you're talking about.

Will add for now but we should discuss the meaning of **the tip of the iceberg** that is release notes ... Reference content on its own likely is not enough to do something, so why not just link to something you can actually use out of the box?

General thoughts on links in release notes, for further discussion:

- **The user journey into a feature should guide you clearly** — It shouldn't start with a bunch of doors you have to choose from at the beginning. That means fewer links with a clear path, not more options.
- **Progressive disclosure tells us to provide reference links in the context where they matter** — i.e., in the notebook itself, where we show you how to use a function or metric.
- **Release notes should feel clean and simple** — Linking maximalism is the antithesis of that. (This might partly be a repeat comment on a recent PR where we added scores of new buttons. I think they add clutter, increase authoring cost, and most will likely never be clicked.)
- **Treat release notes as if they had an expiration date** — Related to the previous point, release notes are like water under the bridge: once something has been published, it receives fewer and fewer eyeballs over time. The real value lies in the core documentation for a feature that sticks around and where people will look for information.
- **Don't forget the authoring cost** — Release notes are expensive to author to begin with, and we should limit how much time we spend on them. Looking up reference content links is fairly low on the bang-for-the-buck scale (and if the release notes are the only place where they can find a link, that's a bug and not a feature).

All of which goes to say: use links sparingly, including in release notes.

nrichers commented:
> Forgot to mention, I like the new rounded corners!

@validbeck I rounded the corners a bit more via df602de03827d7a3b324e382a11f8d6107627a05 which also happens to solve the issue we've been having with corner artefacts from screenshots on macOS. 

| Before | After |
|----------|----------|
| <img width="62" alt="image" src="https://github.com/user-attachments/assets/671fd6b2-1f63-42b1-a4c0-915c73c61e2e">  | <img width="60" alt="image" src="https://github.com/user-attachments/assets/fc9926ff-12b3-43b0-a799-6e1cdfcf5d31"> |

nrichers commented:
> ### GLOBAL COMMENTS
> * **We've being using `target="_blank` for CTA links out of release notes, FYI.** Can you please add these?

Removed from the style guide, where you had add this.

> * **I don't like the "Try it..." or "Learn about..." especially without the name of the link, and as we indicate in our style guide, nebulous/clumsy links should be avoided.** For example, this is what I ended up doing in the clean-up:
>   <img alt="Screenshot 2024-10-21 at 8 59 19 AM" width="888" src="https://private-user-images.githubusercontent.com/164545837/378506386-10378a22-013a-48ec-bfe0-8a518341031c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NDA1NjIsIm5iZiI6MTcyOTU0MDI2MiwicGF0aCI6Ii8xNjQ1NDU4MzcvMzc4NTA2Mzg2LTEwMzc4YTIyLTAxM2EtNDhlYy1iZmUwLThhNTE4MzQxMDMxYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMVQxOTUxMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZjM4NTkwOTU2NjZjZGQ0M2E2NTlkYmZiOTBiZmQ0OGYwMGQyZWRjMTY4ZTMwYTJiMjJlZjdjYWIyZmI2NWY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0_QQd8Rj0F9chHKqlSun93ZDYyzUe0vXd1PcIj6TtjM"> (**If there is a user guide, just use the name of the guide! No need for a lead-in.**)

I replaced all the docs links with the actual topic title and I think that is a generally good rule to follow, thank you. 

I do NOT agree with adding lead-in text for a button that describes what the button does ... That's just a simpler button replaced with text + button. As a general rule, we should be mindful about avoiding unnecessary verbiage — less is more!

Also, we have some notebooks that have very long titles that themselves need to be edited and that would make for some pretty painful buttons:

<img width="1026" alt="image" src="https://github.com/user-attachments/assets/217fdb81-4ccb-413e-abfb-cf3a8e0dd764">

Here's how the buttons now look — doc links use the exact title and notebook links remain as before:

<img width="979" alt="image" src="https://github.com/user-attachments/assets/c4f7dfc1-1e99-4c26-98ec-11f40243d439">

> * **I also don't like the FA icons, too busy.** Please remove them.

Agreed, removed.

> * **Video thumbnails could be bigger.** You can steal the dimensions from this release: https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6860/release-notes-to-sprint-60/releases/2024-may-22/release-notes.html#new-quickstart-video

They're the exact same size as in the docs ... I think if people want a bigger view, they can just watch them on YouTube.

> * I would break up some of the more chunkier paragraphs into some bullet points.

I scanned the text and searched for paragraphs of >200 words. I find release notes often look fragmented anyway, so I made no changes here but if there are specific sections where you think this would help, go ahead. 

> * The `_how-to-upgrade.qmd` includes is in `main`, so you might need to pull in to retrieve it so you can append it at the end.

Yup, that fixed it! Not sure why it was missing when I created the release notes initially but the file is there now. 

> ### Section comments
> #### Support for ongoing monitoring
> * The screenshot here needs to be cropped ;)

I would normally agree but in this case there is a new "Ongoing Monitoring" option in the navigation on the left that you would not see if you cropped the screenshot. (FWIW, the docs for ongoing monitoring use only a very cropped version that has been reduced to the graph.)

> * I'd also do a two column for the image with all the text up to "The monitoring template for your model automatically populates with data as your code runs..." and see what that looks like.

Since the screenshot is not being cropped, I'm going to leave this as is. I think it's fine to have a full-width screenshot here as ongoing monitoring is basically our halo feature in these release notes. 

> #### Support for bias and fairness tests
> We need some links to the metric test reference pages!

Links added. 

> #### New Metric Over Time content block
> The three screenshots are two cramped. I'd try two rows:
> 
> 1. Text | Adding a new block & CTA link (button should be centered)
> 2. Selecting the block | Metric over time graph

Agreed, looks better in two columns, with a bit of padding between them. 

Coincidentally, I noticed that you've been adding this in some places: 

```
::: {.w-10-ns}
<!-- FILLER COLUMN DO NOT REMOVE -->
:::
```

These spacer columns should not be necessary with Tachyons CSS, you can use  `.pr1` or `.pl1` to pad columns as needed, see https://tachyons.io/docs/layout/spacing/.

> #### Filtering columns when running tests
> Needs a link to the `DatasetDescription` reference.

Link added. 

> #### Enhanced custom field management with Python support
> This video needs the styling we apply to all videos.

Ugh, done. I sense another CSS class ... manually adding video styling is getting old fast!

> #### Rich text editor for validation guidelines
> The columns here are squished.

As in there's not enough margin? I did add `.pr4` to the left column but if there's something else you meant, let me know. 

> #### Updated training content
> * Let's use the listing standard here to make the training stand out a bit instead of the random screenshot link:  https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6860/release-notes-to-sprint-60/releases/2024-may-22/release-notes.html#sandbox-getting-started

I don't know what this comment means? It's not a screenshot, it's the live training page embedded in the release notes. I did look at the example, but it's just a small callout. I'd prefer to give more emphasis to our training page here. 

> * The column widths could also be adjusted to be a bit neater.

Fixed.
